File: 프로그래머스/2/42842. 카펫/카펫.swift
import Foundation

// [가로, 세로]
// b: 8 ~ 5000
// y: 1 ~ 200만
// 정사각형이거나 가로로 길다.


// brown + yellow가 나눠떨어지는 [가로, 세로]
// 약수를 찾아서
// 2x + 2y - 4 == brown && x >= y 를 체크

// 1. 

func solution(_ brown:Int, _ yellow:Int) -> [Int] {
    
    let N = brown + yellow
    let possible = (1...N).filter { N % $0 == 0 }
    
    for p in possible {
        
        let x = p
        let y = N / p
        
        if x < y { continue }
        
        if 2 * x + 2 * y - 4 == brown { return [x, y] }
    }
    
    return []
}

File: 프로그래머스/2/42842. 카펫/카펫.py
# 브라운, 노랑의 개수알때> 가로 세로를 리턴
# 5000 * 200만
# a * b = brown + yellow

def solution(brown, yellow):
    
    w = 3
    h = 0
    
    while w <= brown + yellow / 3:
        
        h = (brown + yellow) // w
        
        if (w + h) * 2 - 4 == brown and w >= h:
            return [w, h]
        
        else:
            w += 1
        
    return []